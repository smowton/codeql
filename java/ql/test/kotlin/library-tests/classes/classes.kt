
class ClassOne { }

class ClassTwo (val arg: Int) {
    val x: Int = 3
}

abstract class ClassThree {
    abstract fun foo(arg: Int)
}

class ClassFour: ClassThree() {
    override fun foo(arg: Int) {
    }
}

