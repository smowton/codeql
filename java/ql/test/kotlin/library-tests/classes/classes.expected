| classes.kt:0:0:0:0 | ClassesKt | ClassesKt |
| classes.kt:2:1:2:18 | ClassOne | ClassOne |
| classes.kt:4:1:6:1 | ClassTwo | ClassTwo |
| classes.kt:8:1:10:1 | ClassThree | ClassThree |
| classes.kt:12:1:15:1 | ClassFour | ClassFour |
| classes.kt:17:1:18:1 | ClassFive | ClassFive |
| classes.kt:28:1:30:1 | ClassSix | ClassSix |
| classes.kt:34:1:47:1 | ClassSeven | ClassSeven |
| classes.kt:49:1:51:1 | Direction | Direction |
| classes.kt:53:1:57:1 | Color | Color |
| classes.kt:63:1:91:1 | Class1 | Class1 |
| classes.kt:66:20:66:54 | (no string representation) | <anonymous class> |
| classes.kt:68:20:68:74 | (no string representation) | <anonymous class> |
| classes.kt:72:16:77:10 | (no string representation) | <anonymous class> |
| classes.kt:75:24:75:33 | (no string representation) | <anonymous class> |
| classes.kt:81:16:81:38 | (no string representation) | <anonymous class> |
| classes.kt:85:16:85:25 | (no string representation) | <anonymous class> |
| classes.kt:89:16:89:44 | (no string representation) | <anonymous class> |
